### PR TITLE
docs: formalize architectural design and roadmap for ixgbe PoC

### PR DESCRIPTION
**Pull request by 0xfka**
successor of #7 

### README.rst — Expanded architecture & roadmap
The README was rewritten from a short overview into a multi-section architectural document and project roadmap. New and expanded content includes:
- Direct Register Manipulation: explicit treatment of low-level NIC register access and assumptions about hardware state.
- Defined Hardware Behavior & Transparent Documentation: spelled-out hardware expectations, trade-offs, and where the PoC relies on specific NIC behaviors.
- Simplified Internal Logic: rationale for keeping driver logic minimal compared to full kernel drivers.
- Architectural Design Principles: formalized design goals including "Radical Cropping" (minimalist design) and "Inlining Workload with Driver" (performance-oriented placement of work).
- Roadmap / TO‑DOs: structured checklist and next steps for further development and validation.

The document is narrative and design-focused; it clarifies intent, constraints, and next steps for contributors.

### docs/error_handling.rst — Minor formatting
A single blank-line whitespace change was added for readability.

### Impact, risks, and recommended next steps
- Scope: Documentation-only; no code or runtime behavior changed (net +60 / -6 lines).
- Risks: The README documents several low-level hardware assumptions and aggressive design choices that could introduce correctness or safety risks if implemented without validation (race conditions, memory ordering, interrupt handling, NIC state transitions). These concerns are not addressed in this PR.
- Recommendations: Before productionizing design choices, add: explicit concurrency/ordering notes, test vectors for hardware edge cases, integration tests exercising register-level failure modes, and a compatibility matrix for supported NIC firmware/hardware revisions.

This PR formalizes the PoC architecture and exposes open items to be addressed in subsequent work.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,8 +1,62 @@
...
-.. |docs| image:: /docs/
+Direct Register Manipulation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Instead of sending commands to "mailbox[1]", direct register manipulation considered more transparent when it comes to understand & optimize on low level.
+As a trade-off, even this may mean compromising latency, 82599 is preferred because modern NICs possess multilayered abstractions.
+
+Defined hardware behavior & Transparent documentation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+82599 Datasheet is crystal clear, and potential edge cases are documented over the years. As mentioned in previous title, even initializing a 82599 ASIC vibes like playing with electrons manually.
+Similar attempt on a modern NIC would feel like sending mails to 'mailbox', since it is.
```
[the remark this answers is not in the document]
Comment (fatih881) [reply README.rst]: Let's stay human.
